adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -109,47 +109,25 @@ - (void)presentFromViewController:(nonnull UIViewController *)viewController {
     return;
   }
   [_adEventDelegate willPresentFullScreenView];
-  [UnityAds addDelegate:self];
-  [UnityAds show:viewController placementId:_placementID];
+  [UnityAds show:viewController placementId:_placementID showDelegate:self];
 }
 
 #pragma mark - UnityAdsExtendedDelegate Methods
 
 - (void)unityAdsDidError:(UnityAdsError)error withMessage:(nonnull NSString *)message {
-  [UnityAds removeDelegate:self];
-  if (_adEventDelegate) {
-    NSError *errorWithDescription =
-        GADMAdapterUnitySDKErrorWithUnityAdsErrorAndMessage(error, message);
-    [_adEventDelegate didFailToPresentWithError:errorWithDescription];
-  }
+  // Logic to handle error events has moved to the UnityAdsShowDelegate function
```

Comment:
I'm might be out of the context but this a bit smelly :( 
Is there is a reason why our delegate doesn't do anything here? should we conform to this protocol? and do nothing? maybe logs that is deprecated?

Reply:
This should not be called since `UnityAdsExtendedDelegate` was removed from the class, and `AddDelegate` is no longer getting called. My understanding is this was only handling show errors, and so all the logic could be moved to `UnityAdsShowDelegate`'s `unityAdsShowFailed` method.  (I think between the init, load and show delegates, the extended delegate is no longer needed, so this was a half step to remove it completely from the adapter)

If I am correct that they are not needed at all, then we can try just deleting all the methods for `UnityAdsExtendedDelegate` and see if the change is accepted.

Reply:
Thank you for clarifying. 👍

Reply:
@richardh-unity3d yes, I think we can safely delete these delegates now